### PR TITLE
Add SSE blog

### LINKED ISSUE #44: Blog: What's SSE and how to use it

### PR DESCRIPTION
**Pull request by xxchan**
used https://github.com/aos-dev/go-storage-example/pull/4

Close #44

### CODE REVIEW

File: blog/2021-04-27-sse.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+tags: [updates]
+---
+
+Data protection includes protecting data while in transit (as it travels to and from the server) and at rest (while it is stored on disk). You can do the following to protect your data when using storage services:
+
+- **Secure Socket Layer/Transport Layer Security (SSL/TLS)**: Protect data in transit.
+- **Client-Side Encryption**: You manage the encryption keys, encryption process, and related tools all yourself. This can protect data both in transit and at rest, and this is transparent to the server.
+- **Server-Side Encryption**: Request the server to encrypt your data before it is persisted to disk. 
+
+Recently, we [implemented SSE support](https://github.com/aos-dev/go-storage/issues/523) in [go-storage], and let's talk about SSE here.
```
Comment (Xuanwo) [inline blog/2021-04-27-sse.md:14]: How about adding a notice for abbr `SSE` like `about Server-Side Encryption (shorten to SSE) here`?

File: blog/2021-04-27-sse.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+### SSE with Customer-Provided Keys (SSE-C)
+
+In this way, you create and store encryption keys yourself. Different from Client-Side Encryption, in SSE-C you send unencrypted data along with the encryption key to the server, and then the server uses the key to encrypt your data, but doesn't store the key. When reading an encrypted object, you will have to carry the encryption key again to let the server decrypt your data with the key. In other words, you manage the encryption keys, and the server manages the encryption process and related tools.
+
+## How to Use SSE in go-storage
+
+Since different services support different SSE options and have different behaviors, SSE-related pairs are considered service pairs. So you will have to check which options are provided for your specific service first. We listed supported options and related pairs in our [service docs]. You'd better first choose the SSE option you want to use, and then ignore other pairs in case of being confused with the usage of the pairs.
+
+Take [s3] for example. The code snippets come from [go-storage-example]
```
Comment (Xuanwo) [inline blog/2021-04-27-sse.md:42]: Let's use an example that uses `DefaultParis`, in this way, the user only provides SSE-related options during `NewStorager` and can write the same logic for different services.

As described in https://aos.dev/docs/go-storage/pairs/index#default-pairs

Comment (xxchan) [reply blog/2021-04-27-sse.md:42]: Updated in https://github.com/aos-dev/go-storage-example/pull/4  
PTAL

Comment (Xuanwo) [reply blog/2021-04-27-sse.md:42]: The example looks good to me.

**Comment by xxchan** [comment]
links not fixed

**Comment by xxchan** [comment]
ready now

**Review by Xuanwo — APPROVED** [review]
Great Work!

**Pull request merged**